pyproject.toml

```
@@ -0,0 +1,128 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "reprostim"
+dynamic = ["version"]
+description = 'ReproStim is a video capture and recording suite for neuroimaging and psychology experiments.'
+readme = "README.md"
+#requires-python = "3.10"
+license = "MIT"
+license-files = { paths = ["LICENSES/MIT.txt"] }
+keywords = [
+  "ReproStim",
+  "reprostim-videocapture",
+  "reprostim-screncapture",
+]
+authors = [
+  { name = "Yaroslav Halchenko", email = "yoh@dartmouth.edu" },
+  { name = "Andy Connolly", email = "andrew.c.connolly@dartmouth.edu" },
+  { name = "Horea Christian", email = "chr@chymera.eu" },
+  { name = "Vadim Melnik", email = "vmelnik@docsultant.com" },
+]
+classifiers = [
+  "Development Status :: 3 - Alpha",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+
+dependencies = [
+  "click>=8.1.7",
+  "click-didyoumean>=0.3.1",
+  "pydantic>=2.7.1",
+  "numpy>=1.26.4",
+  "pyzbar>=0.1.9",
+  "qrcode>=8.0",
+  "opencv-python>=4.9.0.80",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest>=8.0",
+]
+audio = [
+  "sounddevice>=0.5.1",
+  "scipy>=1.14.1",
+  "pydub>=0.25.1",
+  "pyaudio>=0.2.14",
+  "reedsolo>=1.7.0",
+]
+psychopy = [
+  "psychopy",
+  "psychopy-sounddevice",
+]
+all = [
+  "pytest>=8.0",
+  "sounddevice>=0.5.1",
+  "scipy>=1.14.1",
+  "pydub>=0.25.1",
+  "pyaudio>=0.2.14",
+  "reedsolo>=1.7.0",
+  "psychopy",
+  "psychopy-sounddevice",
+]
```

This can be done using the done by referencing the project in the specification of the feature group/extra.

```suggestion
all = [
  "reprostim[test]",
  "reprostim[audio]",
  "reprostim[psychopy]",
] 
```

If you take my above suggestion about removing the `test` feature group and specifying it in a dedicated test environment, remove the line of `"reprostim[test]",` from the suggestion.